protocol/tb.go

```diff
@@ -40,5 +40,5 @@ func (tb *TemporaryBinding) Serialize(strSig []byte) []byte {
 // the existed binding (TOFU).
 func (tb *TemporaryBinding) Verify(index, value []byte) bool {
 	return bytes.Equal(tb.Index, index) &&
-		(value != nil && bytes.Equal(tb.Value, value))
+		(value == nil || (value != nil && bytes.Equal(tb.Value, value)))
```

Comment:
Ok, I see what's happening here now.

In `verifyAuthPath` we do,
https://github.com/coniks-sys/coniks-go/blob/master/protocol/consistencychecks.go#L225-L229

Maybe we should do the same in `verifyReturnedPromise`?